pilot/pkg/kube/inject/inject.go

```diff
@@ -639,6 +640,13 @@ func intoObject(sidecarTemplate string, meshconfig *meshconfig.MeshConfig, in ru
 	}
 	metadata.Annotations[annotationStatus.name] = status
 
+	if metadata.Labels == nil {
+		metadata.Labels = make(map[string]string)
+	}
+	// We label it with istio mtls ability to true since it's deployed with Envoy sidecar. Because
+	// it can accept mtls traffic.
```

Comment:
It should still be configurable - we can't add the 'mtls=true' blindly on each inject. 

1. Check if user didn't already set the label ( maybe to false if he doesn't want that pod to use tls)
2. Some option on both sidecar injector and kube-inject to allow disabling it

In general, the rule is that any feature should have a flag and user should be able to control it ( a-la-carte), in
particular for those enabled by default.

Reply:
This is tricky. This label is really NOT configuration, it is a property of the workload. If we could automatically discern this on the data path, then we would not use this mechanism.

What that means is that if a workload has istio sidecar injected, we know that it is capable of doing istio-mtls.

User already has configuration in destination rules to not use mtls while talking to this workload / service.

Consider a scenario where I have a deployment with istio-mtls:false because my deployment does not support mtls. If now I sidecar inject this deployment, I would want that flag to update to reflect that now it does support mtls.

Reply:
Also should we make this annotation instead? Label is going to be indexed by the api server.

Reply:
Please make this annotation. Also, please refer to others for the naming these things :(. istio_workload_mtls_ability... is rather long and convoluted..
`sidecar.istio.io/enabled: true` is more generic and useful for a variety of purposes..

Reply:
@costinm   @mandarjog 
- I changed to `authentication.istio.io/able_mtls` as annotation name. Not `*sidecar*` since as @costinm mentioned this is not about sidecar per se. Workloads without sidecar able to speak mTLS can have this annotation as well.
- We do have to annotate more than Pod, otherwise things like `Deployment` won't get annotation.
- Regarding flag control, I'll add another flag on whether Pilot to configure client TLS settings based on these annotations. I think that'll be enough.